src/client/Microsoft.Identity.Client/Platforms/Android/AndroidTokenCacheAccessor.cs

```diff
@@ -79,25 +82,28 @@ public void SaveAccount(MsalAccountCacheItem item)
             editor.PutString(item.GetKey().ToString(), item.ToJsonString());
             editor.Apply();
         }
+        #endregion
 
-        public void DeleteAccessToken(MsalAccessTokenCacheKey cacheKey)
+        #region DeleteItem
+        public void DeleteAccessToken(MsalAccessTokenCacheItem item)
         {
-            Delete(cacheKey.ToString(), _accessTokenSharedPreference.Edit());
+            Delete(item.GetKey().ToString(), _accessTokenSharedPreference.Edit());
         }
 
-        public void DeleteRefreshToken(MsalRefreshTokenCacheKey cacheKey)
+        public void DeleteRefreshToken(MsalRefreshTokenCacheItem item)
         {
-            Delete(cacheKey.ToString(), _refreshTokenSharedPreference.Edit());
+            Delete(item.GetKey().ToString(), _refreshTokenSharedPreference.Edit());
         }
 
-        public void DeleteIdToken(MsalIdTokenCacheKey cacheKey)
+        public void DeleteIdToken(MsalIdTokenCacheItem item)
         {
-            Delete(cacheKey.ToString(), _idTokenSharedPreference.Edit());
+            Delete(item.GetKey().ToString(), _idTokenSharedPreference.Edit());
```

Comment:
null checks on all the .GetKey().ToString()?

Reply:
We should add pre-conditions or Debug.Assert instead, sadly we don't have nullable references. But item cannot be null and getkey cannot be null.